lib/routes/shared/security_pin/prompt_pin_code.dart

```diff
@@ -0,0 +1,174 @@
+import 'package:breez/bloc/blocs_provider.dart';
+import 'package:breez/bloc/user_profile/breez_user_model.dart';
+import 'package:breez/bloc/user_profile/user_actions.dart';
+import 'package:breez/bloc/user_profile/user_profile_bloc.dart';
+import 'package:breez/theme_data.dart' as theme;
+import 'package:breez/widgets/back_button.dart' as backBtn;
+import 'package:breez/widgets/pin_code_widget.dart';
+import 'package:breez/widgets/route.dart';
+import 'package:flutter/material.dart';
+
+const PIN_CODE_LENGTH = 6;
+
+class LockScreen extends StatefulWidget {
+  final String label;
+  final bool dismissible;
+  final bool changePin;
+  final Widget route;
+
+  LockScreen({Key key, this.label, this.dismissible = false, this.changePin = false, this.route}) : super(key: key);
+
+  @override
+  _LockScreenState createState() => new _LockScreenState();
+}
+
+class _LockScreenState extends State<LockScreen> {
+  UserProfileBloc _userProfileBloc;
+  BreezUserModel _user;
+  String _label;
+
+  String _enteredPinCode = "";
+  String _errorMessage = "";
+  String _tmpPinCode = "";
+
+  bool _isInit = false;
+
+  @override
+  void initState() {
+    super.initState();
+    _setLabel(widget.label ?? "Enter your PIN");
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    if (!_isInit) {
+      _userProfileBloc = AppBlocsProvider.of<UserProfileBloc>(context);
+      _userProfileBloc.userStream.listen((user) {
+        _user = user;
+      });
+      _isInit = true;
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return WillPopScope(
+      onWillPop: () => Future.value(false),
+      child: Scaffold(
+        appBar: widget.dismissible
+            ? new AppBar(
+                iconTheme: theme.appBarIconTheme,
+                textTheme: theme.appBarTextTheme,
+                backgroundColor: theme.BreezColors.blue[500],
+                leading: backBtn.BackButton(
+                  onPressed: () {
+                    Navigator.pop(context, false);
+                  },
+                ),
+                elevation: 0.0,
+              )
+            : null,
+        body: PinCodeWidget(
+          _label,
+          _enteredPinCode,
+          widget.dismissible,
+          _errorMessage,
+          (numberText) => _onNumButtonPressed(numberText),
+          (enteredPinCode) => _setPinCodeInput(enteredPinCode),
+        ),
+      ),
+    );
+  }
+
+  _onNumButtonPressed(String numberText) {
+    _setErrorMessage("");
+    if (_enteredPinCode.length < PIN_CODE_LENGTH) {
+      _setPinCodeInput(_enteredPinCode + numberText);
+    }
+    if (_enteredPinCode.length == PIN_CODE_LENGTH) {
+      if ((widget.changePin) && _tmpPinCode.isEmpty) {
```

Comment:
I can see now the next simplification:
I think it is more readable to handle each use case in its own logical block:
```suggestion
      if ((widget.changePin) {
         _handleChangePin()
      } else {
         _handleUnlockPin()
      }
      
      void _handleChangePin(){
         if(_tmpPinCode.isEmpty) {
            // set tmpPinCode
         } else {
            // validate change pin code
         }
      }
```